File: cherry_pick2.js
// 配列nsからx回、逐次的にその時点での最大のnを１減らした配列にするメソッド
// for while 禁止

const cherry_pick = function(ns, x){
    let i=1;
    function minus() {
        if(i>x){ return; }
        mx = ns.indexOf(Math.max.apply(null,ns));
        ns[mx] -= 1;
        i++;
        minus();
    }
    minus();
    return ns;
}

console.log(cherry_pick([9, 7, 8], 3));
console.log(cherry_pick([9, 7, 8], 10));

File: cherry_pick2.py
# 配列nsからx回、逐次的にその時点での最大のnを１減らした配列にするメソッド
# for while禁止




def cherry_pick(ns, x):
    if (x<1):
        return print(ns)
    else:
        mx = ns.index(max(ns))
        ns[mx] -= 1
        x -= 1
        cherry_pick(ns,x)
    

    

cherry_pick([9, 7, 8], 3)
cherry_pick([9, 7, 8], 10)

